src/parse.c

```diff
@@ -1215,6 +1215,12 @@ handle_bridge_interfaces(NetplanParser* npp, yaml_node_t* node, const void* data
                 g_debug("%s: Bridge contains openvswitch interface, choosing OVS backend", npp->current.netdef->id);
                 npp->current.netdef->backend = NETPLAN_BACKEND_OVS;
             }
+            /* If the bridge has a OVS backend then any gretap or ip6gretap tunnels attached to the bridge must also be OVS */
+            if (component->type == NETPLAN_DEF_TYPE_TUNNEL &&
+                    (component->tunnel.mode == NETPLAN_TUNNEL_MODE_GRETAP || component->tunnel.mode == NETPLAN_TUNNEL_MODE_IP6GRETAP) &&
+                    npp->current.netdef->backend == NETPLAN_BACKEND_OVS) {
+                component->backend = NETPLAN_BACKEND_OVS;
```

Comment:
The netplan net definitions are parsed in the order they are listed in the YAML file.

So if a tunnel is defined before the bridge this code should work. If the bridge is defined first a new netdef will be created via `add_missing_node()`, but that will not necessarily have all the tunnel mode settings ready.

So I think we need an additional check in `handle_tunnel_mode()` to check if this tunnel is an interface to an OVS bridge and then set the `NETPLAN_BACKEND_OVS` accordingly.

We should add test for both cases (bridge first, tunnel first) to make sure this works as expected.